### PR TITLE
Navigation implementation

### PR DESCRIPTION
**Pull request by jchiversAND**
# Description

Implemented React Navigation BottomTabNavigator for navigating between main pages. Included Context additions for the Auth and Login page basic creation to fit navigation flow.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] User testing within simulator, web and iPhone+Android devices.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mobile/src/styles/components/Button.js [hunk 1]
```diff
@@ -21,10 +25,12 @@ const StyledButton = styled(PaperButton)`
   }};
 `;
 
-const Button = ({ children, mode = 'contained', color, onPress = () => {}, ...props }) => {
+const Button = ({ children, mode = 'contained', height, ...props }) => {
   return (
-    <StyledButton mode={mode} color={color} onPress={onPress} {...props}>
-      {children}
+    <StyledButton uppercase={false} mode={mode} {...props}>
+      <Text color={colours.white} fontWeight={font.bold} fontSize={font.small}>
```
[the remark this answers is not in the document]
Comment (jchiversAND) [reply mobile/src/styles/components/Button.js:31]: Silly codeclimate

File: auth/src/app.ts [hunk 2]
```diff
@@ -12,11 +12,12 @@ import { signoutRouter } from './routes/signout';
 import { signupRouter } from './routes/signup';
 
 const app = express();
-app.set('trust proxy', true);
+// app.set('trust proxy', true);
```
Comment (adamjamesturner93) [inline auth/src/app.ts:15]: Why have we commented this out?

File: mobile/App.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 export default function App() {
   return (
-    <PaperProvider theme={PaperTheme}>
-      <NavigationContainer theme={NavigationTheme}>
-        <MainStackNavigation />
-      </NavigationContainer>
-    </PaperProvider>
+    <AuthProvider>
+      <PaperProvider theme={PaperTheme}>
+        <SafeAreaView style={{ flex: 0, backgroundColor: colours.yellowLighter }} />
```
Comment (adamjamesturner93) [inline mobile/App.js:14]: What is the empty SafeAreaView for?

Comment (adamjamesturner93) [inline mobile/App.js:14]: Can we move the inline styles to a UI component?

File: mobile/src/contexts/ActivityContext/ActivityContext.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+const ActivityContext = React.createContext();
+
+export const ActivityProvider extends React.Component {
```
Comment (adamjamesturner93) [inline mobile/src/contexts/ActivityContext/ActivityContext.js:5]: Can we make this functional, same as the AuthContext?

File: mobile/src/navigation/MainStackNavigation/MainStackNavigation.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+function BottomStackNavigation() {
+  return (
+    <Tab.Navigator
+      screenOptions={({ route }) => ({
+        tabBarIcon: ({ focused, color }) => {
+          let iconName;
+
+          if (route.name === 'Leaderboard') {
+            iconName = focused ? 'trophy' : 'trophy';
```
Comment (adamjamesturner93) [inline mobile/src/navigation/MainStackNavigation/MainStackNavigation.js:19]: The `iconName` doesn't change so why do we have a tenary statement?

File: mobile/src/navigation/MainStackNavigation/MainStackNavigation.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import { colours, font } from '../../styles/constants';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { Text } from '../../styles/components';
+
+const Tab = createBottomTabNavigator();
+
+function BottomStackNavigation() {
+  return (
+    <Tab.Navigator
+      screenOptions={({ route }) => ({
```
Comment (adamjamesturner93) [inline mobile/src/navigation/MainStackNavigation/MainStackNavigation.js:14]: For cleanliness can we move this config out of being inline? Just makes it difficult to read :)

File: mobile/src/screens/LoginScreen/LoginScreen.jsx [hunk 7]
```diff
@@ -1,17 +1,37 @@
-import React from 'react';
-import { Text, View } from 'react-native';
-import { Button } from '../../styles/components';
+import React, { useState } from 'react';
+import { Title, Button, Layout, Container, TextInput, Link, Text } from '../../styles/components';
+import { withAuthContext } from '../../contexts/AuthContext/AuthContext';
+import styled from '@emotion/native';
```
Comment (adamjamesturner93) [inline mobile/src/screens/LoginScreen/LoginScreen.jsx:4]: I thought we had moved all the styles into the styles component directory?

File: mobile/src/styles/index.js [hunk 8]
```diff
@@ -1,3 +1,4 @@
 export { default as NavigationTheme } from './themes/NavigationTheme';
 export { default as PaperTheme } from './themes/PaperTheme';
-export { default as ColorTheme } from './constants/ColorScheme';
+export { default as ColorScheme } from './constants/ColourScheme';
```
Comment (adamjamesturner93) [inline mobile/src/styles/index.js:3]: `ColourScheme` please :)

File: auth/src/app.ts (hunk 2 above)
Comment (jchiversAND) [reply auth/src/app.ts:15]: Apparently was you 5 days ago hah. Shall I uncomment it?

File: mobile/App.js (hunk 3 above)
Comment (jchiversAND) [reply mobile/App.js:14]: This allows a separate colour SafeArea at the top (i.e. around the notch on iPhone X +) to the bottom SafeArea below the navigation bar. 

Our designs want it the yellow background around the top, but under the navigation tab bar needs to be the dark blue colour.

File: auth/src/app.ts (hunk 2 above)
Comment (adamjamesturner93) [reply auth/src/app.ts:15]: Oops, leave it then :) I'll address it.

File: mobile/App.js (hunk 3 above)
Comment (jchiversAND) [reply mobile/App.js:14]: Have moved to it's own component, but trying to change from inline to styled components breaks it on iOS but not on web for some reason ?

File: mobile/src/contexts/ActivityContext/ActivityContext.js (hunk 4 above)
Comment (jchiversAND) [reply mobile/src/contexts/ActivityContext/ActivityContext.js:5]: Removed for now as is an unused file currently and is written wrong.

File: mobile/src/navigation/MainStackNavigation/MainStackNavigation.js (hunk 5 above)
Comment (jchiversAND) [reply mobile/src/navigation/MainStackNavigation/MainStackNavigation.js:19]: Good catch, leftover from a good ole copy paste.

File: mobile/App.js (hunk 3 above)
Comment (adamjamesturner93) [reply mobile/App.js]: Hmm, strange? Don't worry if we can't avoid it.

File: mobile/src/screens/LoginScreen/LoginScreen.jsx (hunk 7 above)
Comment (jchiversAND) [reply mobile/src/screens/LoginScreen/LoginScreen.jsx]: Correct, fixed :)

File: mobile/src/styles/components/Button.js [hunk 9]
```diff
@@ -1,14 +1,20 @@
 import React from 'react';
 import { Button as PaperButton } from 'react-native-paper';
-import styled, { css } from '@emotion/native';
-import colors from '../constants/ColorScheme';
+import styled from '@emotion/native';
+import { colours, font } from '../constants';
+import { Text } from '../../styles/components';
```
Comment (adamjamesturner93) [inline mobile/src/styles/components/Button.js:5]: We're already in styles/components aren't we? Not sure why we are going out to come back in?

**Pull request merged**